tabular/src/autogluon/tabular/trainer/model_presets/presets.py

```diff
@@ -112,121 +110,168 @@
+
+    if not silent:
+        logger.log(20, 'Model configs to train:')
     models = []
+    model_args_fit = {}
     for model in model_priority_list:
-        model_type = model[AG_ARGS]['model_type']
-        if not inspect.isclass(model_type):
-            model_type = MODEL_TYPES[model_type]
-        elif not issubclass(model_type, AbstractModel):
-            logger.warning(f'Warning: Custom model type {model_type} does not inherit from {AbstractModel}. This may lead to instability. Consider wrapping {model_type} with an implementation of {AbstractModel}!')
-        else:
-            logger.log(20, f'Custom Model Type Detected: {model_type}')
-        name_orig = model[AG_ARGS].get('name', None)
-        if name_orig is None:
-            name_main = model[AG_ARGS].get('name_main', DEFAULT_MODEL_NAMES.get(model_type, model_type.__name__))
-            name_prefix = model[AG_ARGS].get('name_prefix', '')
-            name_suff = model[AG_ARGS].get('name_suffix', '')
-            name_orig = name_prefix + name_main + name_suff
-        if name_suffix is not None:
-            name_orig = name_orig + name_suffix
-        name = name_orig
-        name_stacker = None
-        num_increment = 2
-        if ensemble_kwargs is None:
-            while name in model_names_set:  # Ensure name is unique
-                name = f'{name_orig}_{num_increment}'
-                num_increment += 1
-            model_names_set.add(name)
-        else:
-            name_bag_suffix = model[AG_ARGS].get('name_bag_suffix', '_BAG')
-            name_stacker = f'{name}{name_bag_suffix}_L{level}'
-            while name_stacker in model_names_set:  # Ensure name is unique
-                name = f'{name_orig}_{num_increment}'
-                name_stacker = f'{name}{name_bag_suffix}_L{level}'
-                num_increment += 1
-            model_names_set.add(name_stacker)
-        model_params = copy.deepcopy(model)
-        model_params.pop(AG_ARGS, None)
-        model_params.pop(AG_ARGS_ENSEMBLE, None)
-        model_init = model_type(path=path, name=name, problem_type=problem_type, eval_metric=eval_metric, stopping_metric=stopping_metric, num_classes=num_classes, hyperparameters=model_params, feature_metadata=feature_metadata)
+        model_init = model_factory(model, path=path, problem_type=problem_type, eval_metric=eval_metric,
+                                   num_classes=num_classes, name_suffix=name_suffix, ensemble_type=ensemble_type, ensemble_kwargs=ensemble_kwargs,
+                                   invalid_name_set=invalid_name_set, level=level, feature_metadata=feature_metadata)
+        invalid_name_set.add(model_init.name)
+        if 'hyperparameter_tune_kwargs' in model[AG_ARGS]:
+            model_args_fit[model_init.name] = {'hyperparameter_tune_kwargs': model[AG_ARGS]['hyperparameter_tune_kwargs']}
+        if not silent:
+            logger.log(20, f'\t{model_init.name}: \t{model}')
```

Comment:
I'd recommend additional more detailed logging statement (at lower logging level) for higher-verbosity users. Since the initial statement talk about the model-configs that will be trained, it could be good at higher verbosity to actually print these configs (ie. their non-default ag_args, ag_args_fit values, non-default hyperparameter-values, etc) before training commences

Reply:
This actually is the detailed log, the variable names were just a bit confusing. Updated. This is only printed if verbosity >= 3.

Example:

```
Model configs that will be trained:
	RandomForestGini: 	{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass'], 'model_type': <class 'autogluon.tabular.models.rf.rf_model.RFModel'>, 'priority': 100}}
	RandomForestEntr: 	{'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass'], 'model_type': <class 'autogluon.tabular.models.rf.rf_model.RFModel'>, 'priority': 100}}
	ExtraTreesGini: 	{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass'], 'model_type': <class 'autogluon.tabular.models.xt.xt_model.XTModel'>, 'priority': 90}}
	ExtraTreesEntr: 	{'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass'], 'model_type': <class 'autogluon.tabular.models.xt.xt_model.XTModel'>, 'priority': 90}}
	KNeighborsUnif: 	{'weights': 'uniform', 'ag_args': {'name_suffix': 'Unif', 'model_type': <class 'autogluon.tabular.models.knn.knn_model.KNNModel'>, 'priority': 80}}
	KNeighborsDist: 	{'weights': 'distance', 'ag_args': {'name_suffix': 'Dist', 'model_type': <class 'autogluon.tabular.models.knn.knn_model.KNNModel'>, 'priority': 80}}
	LightGBM: 	{'ag_args': {'model_type': <class 'autogluon.tabular.models.lgb.lgb_model.LGBModel'>, 'priority': 70}}
	LightGBMXT: 	{'extra_trees': True, 'ag_args': {'name_suffix': 'XT', 'model_type': <class 'autogluon.tabular.models.lgb.lgb_model.LGBModel'>, 'priority': 70}}
	CatBoost: 	{'ag_args': {'model_type': <class 'autogluon.tabular.models.catboost.catboost_model.CatBoostModel'>, 'priority': 60}}
	XGBoost: 	{'ag_args': {'model_type': <class 'autogluon.tabular.models.xgboost.xgboost_model.XGBoostModel'>, 'priority': 55}}
	NeuralNetMXNet: 	{'ag_args': {'model_type': <class 'autogluon.tabular.models.tabular_nn.tabular_nn_model.TabularNeuralNetModel'>, 'priority': 50}}
	NeuralNetFastAI: 	{'ag_args': {'model_type': <class 'autogluon.tabular.models.fastainn.tabular_nn_fastai.NNFastAiTabularModel'>, 'priority': 45}}
	LightGBMCustom: 	{'num_boost_round': 10000, 'num_threads': -1, 'objective': 'binary', 'verbose': -1, 'boosting_type': 'gbdt', 'learning_rate': 0.03, 'num_leaves': 128, 'feature_fraction': 0.9, 'min_data_in_leaf': 5, 'two_round': True, 'seed_value': 0, 'ag_args': {'model_type': <class 'autogluon.tabular.models.lgb.lgb_model.LGBModel'>, 'name_suffix': 'Custom', 'disable_in_hpo': True, 'hyperparameter_tune_kwargs': None, 'priority': 0}}
```


Example with HPO:

```
Model configs that will be trained:
	LightGBM: 	{'ag_args': {'hyperparameter_tune_kwargs': None, 'model_type': <class 'autogluon.tabular.models.lgb.lgb_model.LGBModel'>, 'priority': 70}}
	LightGBM_2: 	{'ag_args': {'hyperparameter_tune_kwargs': (<class 'autogluon.core.scheduler.fifo.FIFOScheduler'>, {'resource': {'num_cpus': 'auto', 'num_gpus': 'auto'}, 'searcher': 'random', 'search_options': {}, 'checkpoint': None, 'resume': False, 'num_trials': None, 'reward_attr': 'validation_performance', 'time_attr': 'epoch', 'visualizer': 'none', 'dist_ip_addrs': []}), 'model_type': <class 'autogluon.tabular.models.lgb.lgb_model.LGBModel'>, 'priority': 70}}
	CatBoost: 	{'ag_args': {'hyperparameter_tune_kwargs': 'bayesopt', 'model_type': <class 'autogluon.tabular.models.catboost.catboost_model.CatBoostModel'>, 'priority': 60}}

```